crobat/src/lib.rs

```diff
@@ -22,13 +22,14 @@ pub struct Crobat {
 }
 
 impl Crobat {
-    pub async fn new() -> Self {
+    pub async fn connect() -> Result<Self> {
         trace!("building crobat client");
         let addr = "https://crobat-rpc.omnisint.io";
-        let conn = Crobat::build_tls_client(addr).await.unwrap();
-
-        Self {
-            client: CrobatClient::new(conn),
+        match Crobat::build_tls_client(addr).await {
+            Ok(conn) => Ok(Self {
+                client: CrobatClient::new(conn),
+            }),
+            Err(e) => Err(e),
```

Comment:
`VitaError::CrobatError` is defined in the other crate and referencing it would create a circular dependency. Is this error handling ok?

Reply:
Oh the code I wrote above was just pseudo code, you could use your original boxed error or that one is fine as well.

Reply:
Excellent pseudo-code :P Let's keep it that way then, probably more useful to get the original error so we know why it can't connect (if it's not intentional!)

Reply:
Yea we can keep the original error then at some point I'll add a proper `VitaError::CrobatError`, rather than the monkey patch I've implemented at the moment:
https://github.com/junnlikestea/vita/blob/322b2783eb1a0e19fbcc71cce0baa815d1fc604e/src/error.rs#L78-L82